rskj-core/src/main/java/co/rsk/peg/StandardMultisigFederation.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.bitcoinj.script.ScriptChunk;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Immutable representation of an RSK Federation in the context of
+ * a specific BTC network.
+ *
+ */
+
+public class StandardMultisigFederation extends Federation {
+
+    public StandardMultisigFederation(
+        List<FederationMember> members,
+        Instant creationTime,
+        long creationBlockNumber,
+        NetworkParameters btcParams) {
+
+        super(members, creationTime, creationBlockNumber, btcParams);
+    }
+
+    @Override
+    public Script getRedeemScript() {
+        if (redeemScript == null) {
+            redeemScript = ScriptBuilder.createRedeemScript(getNumberOfSignaturesRequired(), getBtcPublicKeys());
+        }
+
+        return redeemScript;
+    }
+
+    @Override
+    public int getNumberOfSignaturesRequired() {
+        List<ScriptChunk> standardRedeemScriptChunks = getRedeemScript().getChunks();
+
+        // the threshold of a multisig is the first chunk of the redeemScript
+        // and this fed is a standard multisig
+        ScriptChunk thresholdChunk = standardRedeemScriptChunks.get(0);
+        return Integer.parseInt(thresholdChunk.toString());
```

Comment:
Can't all this be solved by simply calling `getRedeemScript().getNumberOfSignaturesRequiredToSpend`?

Reply:
not really. that also leads us to an stackOverflow error, since the `getRedeemScript` method uses the `getNumberOfSignaturesRequired` one